src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsFetchingSpec.groovy

```diff
@@ -0,0 +1,1189 @@
+    }
+
+    def "PBS should validate fetch.max-file-size-kb from account config"() {
+        given: "Test start time"
+        def startTime = Instant.now()
+
+        and: "Flush metric #metric"
+        def metric = "placeholder"
+        flushMetrics()
+
+        and: "Default BidRequest"
+        def bidRequest = BidRequest.getDefaultBidRequest(APP)
+
+        and: "Account with enabled fetch, fetch.url, maxFileSizeKb in the DB"
+        def account = getAccountWithEnabledFetch(bidRequest.site.publisher.id).tap {
+            config.auction.priceFloors.fetch.maxFileSizeKb = PBSUtils.randomNegativeNumber
+        }
+        accountDao.save(account)
+
+        when: "PBS processes auction request"
+        def response = floorsPbsService.sendAuctionRequest(bidRequest)
+
+        then: "PBS log should error"
+        def logs = floorsPbsService.getLogsByTime(startTime)
+        def floorsLogs = getLogsByText(logs, bidRequest.site.publisher.id)
+        assert floorsLogs.size() == 1
+        assert floorsLogs == ["placeholder"]
+
+        and: "#metric should be update"
+        def metrics = floorsPbsService.sendCollectedMetricsRequest()
+        assert metrics[metric] == 1
+
+        and: "PBS floors validation failure should not reject the entire auction"
+        assert !response.seatbid?.isEmpty()
+    }
+
+    def "PBS should validate fetch.max-rules from account config"() {
+        given: "Test start time"
+        def startTime = Instant.now()
+
+        and: "Flush metric #metric"
+        def metric = "placeholder"
+        flushMetrics()
+
+        and: "Default BidRequest"
+        def bidRequest = BidRequest.getDefaultBidRequest(APP)
+
+        and: "Account with enabled fetch, fetch.url, maxRules in the DB"
+        def account = getAccountWithEnabledFetch(bidRequest.site.publisher.id).tap {
+            config.auction.priceFloors.fetch.maxRules = PBSUtils.randomNegativeNumber
+        }
+        accountDao.save(account)
+
+        when: "PBS processes auction request"
+        def response = floorsPbsService.sendAuctionRequest(bidRequest)
+
+        then: "PBS log should error"
+        def logs = floorsPbsService.getLogsByTime(startTime)
+        def floorsLogs = getLogsByText(logs, bidRequest.site.publisher.id)
+        assert floorsLogs.size() == 1
+        assert floorsLogs == ["placeholder"]
+
+        and: "#metric should be update"
+        def metrics = floorsPbsService.sendCollectedMetricsRequest()
+        assert metrics[metric] == 1
+
+        and: "PBS floors validation failure should not reject the entire auction"
+        assert !response.seatbid?.isEmpty()
+    }
+
+    def "PBS should validate fetch.timeout-ms from account config"() {
+        given: "Pbs with PF configuration with timeoutMs, minTimeoutMs, maxTimeoutMs"
+        def minTimeoutMs = PBSUtils.randomNumber
+        def maxTimeoutMs = PBSUtils.randomNumber + 10
```

Comment:
same comment as above - if randomNumber is generated separately, max could be less than min here.